src/utils/utils.cpp

```diff
@@ -59,16 +103,34 @@ bool openInNativeExtension(const QString &path) {
     //call ShellExecute internally
     return QDesktopServices::openUrl(QUrl::fromLocalFile(path));
 #elif defined(Q_WS_MAC)
-    QProcess open_process;
-    open_process.start(QLatin1String("open"), QStringList(path));
-    open_process.waitForFinished(-1);
-    return open_process.exitCode() == 0;
+    // mac's open program, it will fork to open the file in a subprocess
+    // so we will wait for it to check whether it succeeds or not
+    QProcess subprocess;
+    subprocess.start(QLatin1String("open"), QStringList(path));
+    subprocess.waitForFinished(-1);
+    return subprocess.exitCode() == 0;
 #elif defined(Q_WS_X11)
-    //xdg-open is sufficient
-    QProcess open_process;
-    open_process.start(QLatin1String("xdg-open"), QStringList(path));
-    open_process.waitForFinished(-1);
-    return open_process.exitCode() == 0;
+    // unlike mac's open program, xdg-open won't fork a new subprocess to open
+    // the file will block until the application returns, so we won't wait for it
+    // and we need another approach to check if it works
+
+    // find out if the file can be opened by xdg-open, xdg-mime
+    // usually they are installed in xdg-utils installed by default
+    QString mime_type;
+    if (!getMimeTypeFromXdgUtils(path, &mime_type))
+        return false;
+    // don't open this type of files from xdg-mime
+    if (mime_type == "application/octet-stream")
```

Comment:
Ops, that's bad. But I think it is rare and due to broken mime database from desktop environment. Cleaning `~/.local/share` might help.